packages/gatsby-theme-newrelic/src/utils/page-tracking/gtag.js

```diff
@@ -0,0 +1,69 @@
+import { canTrack } from '../tracking';
+import warning from 'warning';
+import { getGtmConfig } from '../config';
+
+let initialized = false;
+
+const warnAboutNoop = (config) => {
+  warning(
+    config,
+    "You have enabled Google Tag Manager tracking, but do not have it configured. This route change has not been tracked. Please configure the 'googleTagManager' option in gatsby-config.js"
+  );
+  warning(
+    config.src && config.trackingId,
+    "You have enabled Google Tag Manager, but it is misconfigured. This route change has not been tracked. Please configure the 'googleTagManger.src' and 'googleTagManger.trackingId' options in gatsby-config.js"
+  );
+};
+
+const canSendPageView = (config) => config && config.src && config.trackingId;
+
+const sendPageview = ({ location }, themeOptions) => {
+  // wrap inside a timeout to make sure react-helmet is done with its changes (https://github.com/gatsbyjs/gatsby/issues/11592)
+  requestAnimationFrame(() => {
+    requestAnimationFrame(() => {
```

Comment:
Out of curiosity, why did we go with nested `requestAnimationFrame` calls over [`useEffect` and `setTimeout`](https://github.com/nfl/react-helmet/issues/189#issuecomment-698953347)?

Reply:
i wondered the same, but i was just following suit from what we used for Tessen calls in https://github.com/newrelic/gatsby-theme-newrelic/blob/72f6c726888f494ebedf57196ce38ac63a630a4b/packages/gatsby-theme-newrelic/src/utils/page-tracking/tessen.js#L43

Reply:
I might be able to answer this one for you:

We can't use `useEffect` here because we aren't executing this code inside a React component. Hooks only work inside React components. the `onRouteUpdate` Gatsby hooks is just a normal function.

As per the `requestAnimationFrame`, I noticed this in the [`gatsby-plugin-google-gtag`](https://github.com/gatsbyjs/gatsby/blob/54d4721462b9303fed723fdcb15ac5d72e103778/packages/gatsby-plugin-google-gtag/src/gatsby-browser.js#L21-L28) plugin and figured they knew something I didn't 😄.